api/plot/processor.py

```diff
@@ -0,0 +1,11 @@
+from __future__ import annotations
+
+from typing import Any, Dict
+
+class Processor:
```

Comment:
Explain what this is in pydoc. Also add design document that sketches out the architecture.